clang/include/clang/AST/ExprCXX.h

```diff
@@ -5321,6 +5321,58 @@ class BuiltinBitCastExpr final
   }
 };
 
+// Represents an unexpanded pack where the list of expressions are
+// known. These are used when structured bindings introduce a pack.
+class ResolvedUnexpandedPackExpr final
+    : public Expr,
+      private llvm::TrailingObjects<ResolvedUnexpandedPackExpr, Stmt *> {
```

Comment:
```suggestion
      private llvm::TrailingObjects<ResolvedUnexpandedPackExpr, Expr *> {
```

Reply:
The only place you're using this as a Stmt is for children, and the cast that way is 'safe' (and Expr isa Stmt).  This change likely saves a bunch of casting elsewhere.